File: frontend/src/components/Risk/board.ts
// Static Risk-map metadata for the frontend. Mirrors api/services/risk/board.go
// exactly — the Go file is the source of truth for the engine; this file
// drives the 3D globe placement, adjacency hints, and HUD labels.

export type TerritoryID =
  | 'alaska' | 'northwest-territory' | 'alberta' | 'ontario' | 'quebec'
  | 'greenland' | 'western-us' | 'eastern-us' | 'central-america'
  | 'venezuela' | 'brazil' | 'peru' | 'argentina'
  | 'iceland' | 'great-britain' | 'scandinavia' | 'ukraine'
  | 'northern-europe' | 'southern-europe' | 'western-europe'
  | 'north-africa' | 'egypt' | 'east-africa' | 'congo'
  | 'south-africa' | 'madagascar'
  | 'ural' | 'siberia' | 'yakutsk' | 'irkutsk' | 'kamchatka'
  | 'mongolia' | 'japan' | 'china' | 'afghanistan' | 'middle-east'
  | 'india' | 'siam'
  | 'indonesia' | 'new-guinea' | 'western-australia' | 'eastern-australia';

export type ContinentID =
  | 'north-america' | 'south-america' | 'europe'
  | 'africa' | 'asia' | 'australia';

export interface TerritoryDef {
  id: TerritoryID;
  name: string;
  continent: ContinentID;
  lat: number;
  lon: number;
  adjacent: TerritoryID[];
}

export interface ContinentDef {
  id: ContinentID;
  name: string;
  bonus: number;
  color: string; // Mantine palette key for shading owned territories
}

export const CONTINENTS: ContinentDef[] = [
  { id: 'north-america', name: 'North America', bonus: 5, color: 'electricBlue' },
  { id: 'south-america', name: 'South America', bonus: 2, color: 'hotYellow' },
  { id: 'europe', name: 'Europe', bonus: 5, color: 'synthPurple' },
  { id: 'africa', name: 'Africa', bonus: 3, color: 'neonPink' },
  { id: 'asia', name: 'Asia', bonus: 7, color: 'neonGreen' },
  { id: 'australia', name: 'Australia', bonus: 2, color: 'hotYellow' },
];

export const TERRITORIES: TerritoryDef[] = [
  // North America
  { id: 'alaska', name: 'Alaska', continent: 'north-america', lat: 65, lon: -150,
    adjacent: ['northwest-territory', 'alberta', 'kamchatka'] },
  { id: 'northwest-territory', name: 'NW Territory', continent: 'north-america', lat: 70, lon: -110,
    adjacent: ['alaska', 'alberta', 'ontario', 'greenland'] },
  { id: 'alberta', name: 'Alberta', continent: 'north-america', lat: 55, lon: -115,
    adjacent: ['alaska', 'northwest-territory', 'ontario', 'western-us'] },
  { id: 'ontario', name: 'Ontario', continent: 'north-america', lat: 55, lon: -90,
    adjacent: ['northwest-territory', 'alberta', 'western-us', 'eastern-us', 'quebec', 'greenland'] },
  { id: 'quebec', name: 'Quebec', continent: 'north-america', lat: 55, lon: -75,
    adjacent: ['ontario', 'eastern-us', 'greenland'] },
  { id: 'greenland', name: 'Greenland', continent: 'north-america', lat: 75, lon: -40,
    adjacent: ['northwest-territory', 'ontario', 'quebec', 'iceland'] },
  { id: 'western-us', name: 'Western US', continent: 'north-america', lat: 38, lon: -110,
    adjacent: ['alberta', 'ontario', 'eastern-us', 'central-america'] },
  { id: 'eastern-us', name: 'Eastern US', continent: 'north-america', lat: 38, lon: -85,
    adjacent: ['ontario', 'quebec', 'western-us', 'central-america'] },
  { id: 'central-america', name: 'Central America', continent: 'north-america', lat: 17, lon: -90,
    adjacent: ['western-us', 'eastern-us', 'venezuela'] },

  // South America
  { id: 'venezuela', name: 'Venezuela', continent: 'south-america', lat: 8, lon: -65,
    adjacent: ['central-america', 'brazil', 'peru'] },
  { id: 'brazil', name: 'Brazil', continent: 'south-america', lat: -10, lon: -55,
    adjacent: ['venezuela', 'peru', 'argentina', 'north-africa'] },
  { id: 'peru', name: 'Peru', continent: 'south-america', lat: -10, lon: -75,
    adjacent: ['venezuela', 'brazil', 'argentina'] },
  { id: 'argentina', name: 'Argentina', continent: 'south-america', lat: -35, lon: -65,
    adjacent: ['peru', 'brazil'] },

  // Europe
  { id: 'iceland', name: 'Iceland', continent: 'europe', lat: 65, lon: -19,
    adjacent: ['greenland', 'great-britain', 'scandinavia'] },
  { id: 'great-britain', name: 'Great Britain', continent: 'europe', lat: 54, lon: -3,
    adjacent: ['iceland', 'scandinavia', 'northern-europe', 'western-europe'] },
  { id: 'scandinavia', name: 'Scandinavia', continent: 'europe', lat: 62, lon: 15,
    adjacent: ['iceland', 'great-britain', 'northern-europe', 'ukraine'] },
  { id: 'ukraine', name: 'Ukraine', continent: 'europe', lat: 50, lon: 30,
    adjacent: ['scandinavia', 'northern-europe', 'southern-europe', 'ural', 'afghanistan', 'middle-east'] },
  { id: 'northern-europe', name: 'Northern Europe', continent: 'europe', lat: 52, lon: 15,
    adjacent: ['great-britain', 'scandinavia', 'ukraine', 'southern-europe', 'western-europe'] },
  { id: 'southern-europe', name: 'Southern Europe', continent: 'europe', lat: 44, lon: 12,
    adjacent: ['northern-europe', 'ukraine', 'western-europe', 'north-africa', 'egypt', 'middle-east'] },
  { id: 'western-europe', name: 'Western Europe', continent: 'europe', lat: 46, lon: 2,
    adjacent: ['great-britain', 'northern-europe', 'southern-europe', 'north-africa'] },

  // Africa
  { id: 'north-africa', name: 'North Africa', continent: 'africa', lat: 22, lon: 8,
    adjacent: ['brazil', 'western-europe', 'southern-europe', 'egypt', 'east-africa', 'congo'] },
  { id: 'egypt', name: 'Egypt', continent: 'africa', lat: 27, lon: 30,
    adjacent: ['southern-europe', 'north-africa', 'east-africa', 'middle-east'] },
  { id: 'east-africa', name: 'East Africa', continent: 'africa', lat: 5, lon: 35,
    adjacent: ['egypt', 'north-africa', 'congo', 'south-africa', 'madagascar', 'middle-east'] },
  { id: 'congo', name: 'Congo', continent: 'africa', lat: -2, lon: 22,
    adjacent: ['north-africa', 'east-africa', 'south-africa'] },
  { id: 'south-africa', name: 'South Africa', continent: 'africa', lat: -28, lon: 25,
    adjacent: ['congo', 'east-africa', 'madagascar'] },
  { id: 'madagascar', name: 'Madagascar', continent: 'africa', lat: -20, lon: 47,
    adjacent: ['east-africa', 'south-africa'] },

  // Asia
  { id: 'ural', name: 'Ural', continent: 'asia', lat: 60, lon: 60,
    adjacent: ['ukraine', 'siberia', 'china', 'afghanistan'] },
  { id: 'siberia', name: 'Siberia', continent: 'asia', lat: 65, lon: 95,
    adjacent: ['ural', 'yakutsk', 'irkutsk', 'mongolia', 'china'] },
  { id: 'yakutsk', name: 'Yakutsk', continent: 'asia', lat: 65, lon: 130,
    adjacent: ['siberia', 'irkutsk', 'kamchatka'] },
  { id: 'irkutsk', name: 'Irkutsk', continent: 'asia', lat: 58, lon: 110,
    adjacent: ['siberia', 'yakutsk', 'kamchatka', 'mongolia'] },
  { id: 'kamchatka', name: 'Kamchatka', continent: 'asia', lat: 55, lon: 160,
    adjacent: ['yakutsk', 'irkutsk', 'mongolia', 'japan', 'alaska'] },
  { id: 'mongolia', name: 'Mongolia', continent: 'asia', lat: 47, lon: 105,
    adjacent: ['siberia', 'irkutsk', 'kamchatka', 'china', 'japan'] },
  { id: 'japan', name: 'Japan', continent: 'asia', lat: 36, lon: 138,
    adjacent: ['kamchatka', 'mongolia'] },
  { id: 'china', name: 'China', continent: 'asia', lat: 35, lon: 105,
    adjacent: ['ural', 'siberia', 'mongolia', 'afghanistan', 'india', 'siam'] },
  { id: 'afghanistan', name: 'Afghanistan', continent: 'asia', lat: 35, lon: 65,
    adjacent: ['ural', 'ukraine', 'china', 'india', 'middle-east'] },
  { id: 'middle-east', name: 'Middle East', continent: 'asia', lat: 30, lon: 50,
    adjacent: ['ukraine', 'southern-europe', 'egypt', 'east-africa', 'afghanistan', 'india'] },
  { id: 'india', name: 'India', continent: 'asia', lat: 22, lon: 78,
    adjacent: ['china', 'afghanistan', 'middle-east', 'siam'] },
  { id: 'siam', name: 'Siam', continent: 'asia', lat: 13, lon: 100,
    adjacent: ['china', 'india', 'indonesia'] },

  // Australia
  { id: 'indonesia', name: 'Indonesia', continent: 'australia', lat: -3, lon: 115,
    adjacent: ['siam', 'new-guinea', 'western-australia'] },
  { id: 'new-guinea', name: 'New Guinea', continent: 'australia', lat: -5, lon: 142,
    adjacent: ['indonesia', 'western-australia', 'eastern-australia'] },
  { id: 'western-australia', name: 'Western Australia', continent: 'australia', lat: -25, lon: 122,
    adjacent: ['indonesia', 'new-guinea', 'eastern-australia'] },
  { id: 'eastern-australia', name: 'Eastern Australia', continent: 'australia', lat: -25, lon: 145,
    adjacent: ['new-guinea', 'western-australia'] },
];

const byId = new Map(TERRITORIES.map((t) => [t.id, t]));

export function territoryById(id: TerritoryID): TerritoryDef | undefined {
  return byId.get(id);
}

export function territoriesByContinent(c: ContinentID): TerritoryDef[] {
  return TERRITORIES.filter((t) => t.continent === c);
}

export function adjacent(a: TerritoryID, b: TerritoryID): boolean {
  const t = byId.get(a);
  return !!t && t.adjacent.includes(b);
}

export function continentById(id: ContinentID): ContinentDef | undefined {
  return CONTINENTS.find((c) => c.id === id);
}

/** Convert (lat, lon) on a unit sphere to a 3D position. Used by the globe. */
export function latLonToVec3(lat: number, lon: number, radius = 1): [number, number, number] {
  const phi = (90 - lat) * (Math.PI / 180);
  const theta = (lon + 180) * (Math.PI / 180);
  const x = -radius * Math.sin(phi) * Math.cos(theta);
  const y = radius * Math.cos(phi);
  const z = radius * Math.sin(phi) * Math.sin(theta);
  return [x, y, z];
}

File: api/services/risk/board.go
package risk

// This file is the source of truth for the static Risk board: 42 territories
// in 6 continents and the adjacency graph that ties them together.
//
// Adjacencies are the classic Hasbro rules — including the wrap-around
// trans-oceanic connections Alaska↔Kamchatka, Brazil↔North Africa,
// Western/Southern Europe↔North Africa, Southern Europe↔Egypt, East Africa↔
// Middle East, and Indonesia↔Siam. The adjacency map below is symmetric (an
// edge is encoded on both ends) and is validated by board_test.go.

// TerritoryDef describes one territory's static metadata. Adjacency is the
// list of every TerritoryID that shares a border (including ocean crossings).
type TerritoryDef struct {
	ID        TerritoryID
	Name      string
	Continent ContinentID
	Lat       float64 // approximate geographic centroid for the 3D globe
	Lon       float64
	Adjacent  []TerritoryID
}

// ContinentDef describes one continent's static metadata.
type ContinentDef struct {
	ID    ContinentID
	Name  string
	Bonus int // armies awarded per turn for fully controlling it
}

// Continent IDs.
const (
	ContinentNA   ContinentID = "north-america"
	ContinentSA   ContinentID = "south-america"
	ContinentEU   ContinentID = "europe"
	ContinentAF   ContinentID = "africa"
	ContinentAS   ContinentID = "asia"
	ContinentAUS  ContinentID = "australia"
)

// Continents is the 6-element static continent table.
var Continents = []ContinentDef{
	{ID: ContinentNA, Name: "North America", Bonus: 5},
	{ID: ContinentSA, Name: "South America", Bonus: 2},
	{ID: ContinentEU, Name: "Europe", Bonus: 5},
	{ID: ContinentAF, Name: "Africa", Bonus: 3},
	{ID: ContinentAS, Name: "Asia", Bonus: 7},
	{ID: ContinentAUS, Name: "Australia", Bonus: 2},
}

// Territory IDs. Lower-case kebab-case so they're stable URL/Firestore keys.
const (
	// North America
	TerrAlaska       TerritoryID = "alaska"
	TerrNWTerritory  TerritoryID = "northwest-territory"
	TerrAlberta      TerritoryID = "alberta"
	TerrOntario      TerritoryID = "ontario"
	TerrQuebec       TerritoryID = "quebec"
	TerrGreenland    TerritoryID = "greenland"
	TerrWesternUS    TerritoryID = "western-us"
	TerrEasternUS    TerritoryID = "eastern-us"
	TerrCentralAm    TerritoryID = "central-america"

	// South America
	TerrVenezuela TerritoryID = "venezuela"
	TerrBrazil    TerritoryID = "brazil"
	TerrPeru      TerritoryID = "peru"
	TerrArgentina TerritoryID = "argentina"

	// Europe
	TerrIceland       TerritoryID = "iceland"
	TerrGreatBritain  TerritoryID = "great-britain"
	TerrScandinavia   TerritoryID = "scandinavia"
	TerrUkraine       TerritoryID = "ukraine"
	TerrNorthernEU    TerritoryID = "northern-europe"
	TerrSouthernEU    TerritoryID = "southern-europe"
	TerrWesternEU     TerritoryID = "western-europe"

	// Africa
	TerrNorthAfrica TerritoryID = "north-africa"
	TerrEgypt       TerritoryID = "egypt"
	TerrEastAfrica  TerritoryID = "east-africa"
	TerrCongo       TerritoryID = "congo"
	TerrSouthAfrica TerritoryID = "south-africa"
	TerrMadagascar  TerritoryID = "madagascar"

	// Asia
	TerrUral        TerritoryID = "ural"
	TerrSiberia     TerritoryID = "siberia"
	TerrYakutsk     TerritoryID = "yakutsk"
	TerrIrkutsk     TerritoryID = "irkutsk"
	TerrKamchatka   TerritoryID = "kamchatka"
	TerrMongolia    TerritoryID = "mongolia"
	TerrJapan       TerritoryID = "japan"
	TerrChina       TerritoryID = "china"
	TerrAfghanistan TerritoryID = "afghanistan"
	TerrMiddleEast  TerritoryID = "middle-east"
	TerrIndia       TerritoryID = "india"
	TerrSiam        TerritoryID = "siam"

	// Australia
	TerrIndonesia        TerritoryID = "indonesia"
	TerrNewGuinea        TerritoryID = "new-guinea"
	TerrWesternAustralia TerritoryID = "western-australia"
	TerrEasternAustralia TerritoryID = "eastern-australia"
)

// Territories is the 42-element static territory table. Adjacencies must be
// symmetric — board_test.go enforces this.
var Territories = []TerritoryDef{
	// North America (5)
	{TerrAlaska, "Alaska", ContinentNA, 65, -150, []TerritoryID{TerrNWTerritory, TerrAlberta, TerrKamchatka}},
	{TerrNWTerritory, "NW Territory", ContinentNA, 70, -110, []TerritoryID{TerrAlaska, TerrAlberta, TerrOntario, TerrGreenland}},
	{TerrAlberta, "Alberta", ContinentNA, 55, -115, []TerritoryID{TerrAlaska, TerrNWTerritory, TerrOntario, TerrWesternUS}},
	{TerrOntario, "Ontario", ContinentNA, 55, -90, []TerritoryID{TerrNWTerritory, TerrAlberta, TerrWesternUS, TerrEasternUS, TerrQuebec, TerrGreenland}},
	{TerrQuebec, "Quebec", ContinentNA, 55, -75, []TerritoryID{TerrOntario, TerrEasternUS, TerrGreenland}},
	{TerrGreenland, "Greenland", ContinentNA, 75, -40, []TerritoryID{TerrNWTerritory, TerrOntario, TerrQuebec, TerrIceland}},
	{TerrWesternUS, "Western US", ContinentNA, 38, -110, []TerritoryID{TerrAlberta, TerrOntario, TerrEasternUS, TerrCentralAm}},
	{TerrEasternUS, "Eastern US", ContinentNA, 38, -85, []TerritoryID{TerrOntario, TerrQuebec, TerrWesternUS, TerrCentralAm}},
	{TerrCentralAm, "Central America", ContinentNA, 17, -90, []TerritoryID{TerrWesternUS, TerrEasternUS, TerrVenezuela}},

	// South America (2)
	{TerrVenezuela, "Venezuela", ContinentSA, 8, -65, []TerritoryID{TerrCentralAm, TerrBrazil, TerrPeru}},
	{TerrBrazil, "Brazil", ContinentSA, -10, -55, []TerritoryID{TerrVenezuela, TerrPeru, TerrArgentina, TerrNorthAfrica}},
	{TerrPeru, "Peru", ContinentSA, -10, -75, []TerritoryID{TerrVenezuela, TerrBrazil, TerrArgentina}},
	{TerrArgentina, "Argentina", ContinentSA, -35, -65, []TerritoryID{TerrPeru, TerrBrazil}},

	// Europe (5)
	{TerrIceland, "Iceland", ContinentEU, 65, -19, []TerritoryID{TerrGreenland, TerrGreatBritain, TerrScandinavia}},
	{TerrGreatBritain, "Great Britain", ContinentEU, 54, -3, []TerritoryID{TerrIceland, TerrScandinavia, TerrNorthernEU, TerrWesternEU}},
	{TerrScandinavia, "Scandinavia", ContinentEU, 62, 15, []TerritoryID{TerrIceland, TerrGreatBritain, TerrNorthernEU, TerrUkraine}},
	{TerrUkraine, "Ukraine", ContinentEU, 50, 30, []TerritoryID{TerrScandinavia, TerrNorthernEU, TerrSouthernEU, TerrUral, TerrAfghanistan, TerrMiddleEast}},
	{TerrNorthernEU, "Northern Europe", ContinentEU, 52, 15, []TerritoryID{TerrGreatBritain, TerrScandinavia, TerrUkraine, TerrSouthernEU, TerrWesternEU}},
	{TerrSouthernEU, "Southern Europe", ContinentEU, 44, 12, []TerritoryID{TerrNorthernEU, TerrUkraine, TerrWesternEU, TerrNorthAfrica, TerrEgypt, TerrMiddleEast}},
	{TerrWesternEU, "Western Europe", ContinentEU, 46, 2, []TerritoryID{TerrGreatBritain, TerrNorthernEU, TerrSouthernEU, TerrNorthAfrica}},

	// Africa (3)
	{TerrNorthAfrica, "North Africa", ContinentAF, 22, 8, []TerritoryID{TerrBrazil, TerrWesternEU, TerrSouthernEU, TerrEgypt, TerrEastAfrica, TerrCongo}},
	{TerrEgypt, "Egypt", ContinentAF, 27, 30, []TerritoryID{TerrSouthernEU, TerrNorthAfrica, TerrEastAfrica, TerrMiddleEast}},
	{TerrEastAfrica, "East Africa", ContinentAF, 5, 35, []TerritoryID{TerrEgypt, TerrNorthAfrica, TerrCongo, TerrSouthAfrica, TerrMadagascar, TerrMiddleEast}},
	{TerrCongo, "Congo", ContinentAF, -2, 22, []TerritoryID{TerrNorthAfrica, TerrEastAfrica, TerrSouthAfrica}},
	{TerrSouthAfrica, "South Africa", ContinentAF, -28, 25, []TerritoryID{TerrCongo, TerrEastAfrica, TerrMadagascar}},
	{TerrMadagascar, "Madagascar", ContinentAF, -20, 47, []TerritoryID{TerrEastAfrica, TerrSouthAfrica}},

	// Asia (7)
	{TerrUral, "Ural", ContinentAS, 60, 60, []TerritoryID{TerrUkraine, TerrSiberia, TerrChina, TerrAfghanistan}},
	{TerrSiberia, "Siberia", ContinentAS, 65, 95, []TerritoryID{TerrUral, TerrYakutsk, TerrIrkutsk, TerrMongolia, TerrChina}},
	{TerrYakutsk, "Yakutsk", ContinentAS, 65, 130, []TerritoryID{TerrSiberia, TerrIrkutsk, TerrKamchatka}},
	{TerrIrkutsk, "Irkutsk", ContinentAS, 58, 110, []TerritoryID{TerrSiberia, TerrYakutsk, TerrKamchatka, TerrMongolia}},
	{TerrKamchatka, "Kamchatka", ContinentAS, 55, 160, []TerritoryID{TerrYakutsk, TerrIrkutsk, TerrMongolia, TerrJapan, TerrAlaska}},
	{TerrMongolia, "Mongolia", ContinentAS, 47, 105, []TerritoryID{TerrSiberia, TerrIrkutsk, TerrKamchatka, TerrChina, TerrJapan}},
	{TerrJapan, "Japan", ContinentAS, 36, 138, []TerritoryID{TerrKamchatka, TerrMongolia}},
	{TerrChina, "China", ContinentAS, 35, 105, []TerritoryID{TerrUral, TerrSiberia, TerrMongolia, TerrAfghanistan, TerrIndia, TerrSiam}},
	{TerrAfghanistan, "Afghanistan", ContinentAS, 35, 65, []TerritoryID{TerrUral, TerrUkraine, TerrChina, TerrIndia, TerrMiddleEast}},
	{TerrMiddleEast, "Middle East", ContinentAS, 30, 50, []TerritoryID{TerrUkraine, TerrSouthernEU, TerrEgypt, TerrEastAfrica, TerrAfghanistan, TerrIndia}},
	{TerrIndia, "India", ContinentAS, 22, 78, []TerritoryID{TerrChina, TerrAfghanistan, TerrMiddleEast, TerrSiam}},
	{TerrSiam, "Siam", ContinentAS, 13, 100, []TerritoryID{TerrChina, TerrIndia, TerrIndonesia}},

	// Australia (4)
	{TerrIndonesia, "Indonesia", ContinentAUS, -3, 115, []TerritoryID{TerrSiam, TerrNewGuinea, TerrWesternAustralia}},
	{TerrNewGuinea, "New Guinea", ContinentAUS, -5, 142, []TerritoryID{TerrIndonesia, TerrWesternAustralia, TerrEasternAustralia}},
	{TerrWesternAustralia, "Western Australia", ContinentAUS, -25, 122, []TerritoryID{TerrIndonesia, TerrNewGuinea, TerrEasternAustralia}},
	{TerrEasternAustralia, "Eastern Australia", ContinentAUS, -25, 145, []TerritoryID{TerrNewGuinea, TerrWesternAustralia}},
}

// territoryByID is the lookup index built at init().
var territoryByID = map[TerritoryID]TerritoryDef{}

// adjacency is the symmetric undirected graph built at init() — adjacency[a]
// is the set of every territory that borders a.
var adjacency = map[TerritoryID]map[TerritoryID]struct{}{}

// continentTerritories[c] is the list of every territory in continent c.
var continentTerritories = map[ContinentID][]TerritoryID{}

func init() {
	for _, t := range Territories {
		territoryByID[t.ID] = t
		continentTerritories[t.Continent] = append(continentTerritories[t.Continent], t.ID)
	}
	for _, t := range Territories {
		if adjacency[t.ID] == nil {
			adjacency[t.ID] = map[TerritoryID]struct{}{}
		}
		for _, n := range t.Adjacent {
			adjacency[t.ID][n] = struct{}{}
			// Symmetrize defensively even though Territories already lists both ends.
			if adjacency[n] == nil {
				adjacency[n] = map[TerritoryID]struct{}{}
			}
			adjacency[n][t.ID] = struct{}{}
		}
	}
}

// Adjacent reports whether territories a and b share a border (or
// trans-oceanic route).
func Adjacent(a, b TerritoryID) bool {
	_, ok := adjacency[a][b]
	return ok
}

// Neighbors returns the territories that border t.
func Neighbors(t TerritoryID) []TerritoryID {
	out := make([]TerritoryID, 0, len(adjacency[t]))
	for n := range adjacency[t] {
		out = append(out, n)
	}
	return out
}

// TerritoryByID returns the static definition of a territory.
func TerritoryByID(id TerritoryID) (TerritoryDef, bool) {
	t, ok := territoryByID[id]
	return t, ok
}

// ContinentOf returns the continent the given territory belongs to.
func ContinentOf(t TerritoryID) ContinentID {
	td, ok := territoryByID[t]
	if !ok {
		return ""
	}
	return td.Continent
}

// TerritoriesIn returns the territories on continent c.
func TerritoriesIn(c ContinentID) []TerritoryID {
	out := make([]TerritoryID, len(continentTerritories[c]))
	copy(out, continentTerritories[c])
	return out
}

// ContinentBonus returns the army bonus for fully controlling continent c.
func ContinentBonus(c ContinentID) int {
	for _, def := range Continents {
		if def.ID == c {
			return def.Bonus
		}
	}
	return 0
}

// AllTerritoryIDs returns every territory ID, ordered as in Territories.
func AllTerritoryIDs() []TerritoryID {
	out := make([]TerritoryID, len(Territories))
	for i, t := range Territories {
		out[i] = t.ID
	}
	return out
}
